packages/server/modules/workspaces/repositories/workspaces.ts

```diff
@@ -35,9 +42,32 @@ import { WorkspaceInviteResourceType } from '@/modules/workspaces/domain/constan
 const tables = {
   streams: (db: Knex) => db<StreamRecord>('streams'),
   workspaces: (db: Knex) => db<Workspace>('workspaces'),
+  workspaceDomains: (db: Knex) => db<WorkspaceDomain>('workspace_domains'),
   workspacesAcl: (db: Knex) => db<WorkspaceAcl>('workspace_acl')
 }
 
+export const getUserDiscoverableWorkspacesFactory =
+  ({ db }: { db: Knex }): GetUserDiscoverableWorkspaces =>
+  async ({ domains, userId }) => {
+    if (domains.length === 0) {
+      return []
+    }
+    return (await tables
+      .workspaces(db)
+      .select('workspaces.id as id', 'name', 'description')
+      .distinctOn('workspaces.id')
+      .join('workspace_domains', 'workspace_domains.workspaceId', 'workspaces.id')
+      .leftJoin(
```

Comment:
I see, thats fine then. I guess I was confused because we don't specify which ACL columns we're picking, so I didn't realize that the role != null check refers to that join